pkg/controllers/clusterresourceplacement/placement_status.go

```diff
@@ -342,3 +342,340 @@ func buildFailedResourcePlacements(work *fleetv1beta1.Work) (isPending bool, res
+		}
+		oldConditions, ok := oldResourcePlacementStatusMap[c.ClusterName]
+		if ok {
+			// update the lastTransitionTime considering the existing condition status instead of overwriting
+			rp.Conditions = oldConditions
+		}
+		meta.SetStatusCondition(&rp.Conditions, scheduledCondition)
+		res, err := r.setResourcePlacementStatusPerCluster(ctx, crp, latestResourceSnapshot, resourceBindingMap[c.ClusterName], &rp)
+		if err != nil {
+			return false, err
+		}
+		for i := range res {
+			switch res[i] {
+			case metav1.ConditionTrue:
+				clusterConditionStatusRes[i][TrueConditionStatus]++
+			case metav1.ConditionFalse:
+				clusterConditionStatusRes[i][FalseConditionStatus]++
+			case metav1.ConditionUnknown:
+				clusterConditionStatusRes[i][UnknownConditionStatus]++
+			}
+		}
+		placementStatuses = append(placementStatuses, rp)
+	}
+	isClusterScheduled := len(placementStatuses) > 0
+
+	for i := 0; i < unscheduledClusterCount && i < len(unselected); i++ {
+		// TODO: we could improve the message by summarizing the failure reasons from all of the unselected clusters.
+		// For now, it starts from adding some sample failures of unselected clusters.
+		var rp fleetv1beta1.ResourcePlacementStatus
+		scheduledCondition := metav1.Condition{
+			Status:             metav1.ConditionFalse,
+			Type:               string(fleetv1beta1.ResourceScheduledConditionType),
+			Reason:             ResourceScheduleFailedReason,
+			Message:            fmt.Sprintf(resourcePlacementConditionScheduleFailedMessageFormat, unselected[i].ClusterName, unselected[i].Reason),
+			ObservedGeneration: crp.Generation,
+		}
+		if unselected[i].ClusterScore != nil {
+			scheduledCondition.Message = fmt.Sprintf(resourcePlacementConditionScheduleFailedWithScoreMessageFormat, unselected[i].ClusterName, unselected[i].ClusterScore, unselected[i].Reason)
+		}
+		meta.SetStatusCondition(&rp.Conditions, scheduledCondition)
+		placementStatuses = append(placementStatuses, rp)
+	}
+	crp.Status.PlacementStatuses = placementStatuses
+
+	if !isClusterScheduled {
+		// TODO special handling when isClusterScheduled is false
+		// Currently it will stop updating the resource related conditions and leave the existing conditions there.
+		return isClusterScheduled, nil
+	}
+
+	for i := RolloutStartedCondition; i < TotalCondition; i++ {
+		if clusterConditionStatusRes[i][UnknownConditionStatus] > 0 {
+			crp.SetConditions(i.UnknownClusterResourcePlacementCondition(crp.Generation, clusterConditionStatusRes[i][UnknownConditionStatus]))
+			break
+		} else if clusterConditionStatusRes[i][FalseConditionStatus] > 0 {
+			crp.SetConditions(i.FalseClusterResourcePlacementCondition(crp.Generation, clusterConditionStatusRes[i][FalseConditionStatus]))
+			break
+		} else {
+			crp.SetConditions(i.TrueClusterResourcePlacementCondition(crp.Generation, clusterConditionStatusRes[i][TrueConditionStatus]))
+		}
+	}
+
+	return isClusterScheduled, nil
```

Comment:
isClusterScheduled is always true by now?

Reply:
depends on line 414. left a TODO on line 435 to handle when isClusterScheduled is false.

Reply:
I don't get it, if isClusterScheduled  is false then it won't reach here, right?

Reply:
oh, yes, misunderstood your comment. when it reaches to line 452, it's always true. I can change to "true" if it's your concern.